src/client/dfuse/ops/readdir.c

```diff
@@ -59,14 +59,17 @@ filler_cb(dfs_t *dfs, dfs_obj_t *dir, const char name[], void *_udata)
 	if (rc)
 		return rc;
 
+
+	if (S_ISFIFO(stbuf.st_mode)) {
+		stbuf.st_mode &= ~S_IFIFO;
+		stbuf.st_mode |= S_IFDIR;
+	}
```

Comment:
so all that means that we will never support fifo over dfuse, right?

Reply:
Yes, but we never had any plans to do that.

Reply:
well my question was what if someone is interested in having this in the future?
do we want to close that door?

Reply:
We've discussed this in the POSIX calls and with the customer several times, nobody has ever expressed an indication that we might want to.

There are several other bits we could use, block device or character files etc that would probably work just as well assuming fuse supports them if you think that would be better.

Reply:
I'm just thinking beyond our current plan and how tied in we would be to this. I guess we could  change dfuse in that case to use something else.
but what i don't like is one could call dfs_open with FIFO and expect a FIFO, but they are not  getting that.. using the dfs API that is. what we are doing here is something specific / handled  only in dfuse
we definitely discussed implementing a block device in the future, so we can't use that..

Reply:
im ok with the patch. i just want to make sure that we understand what this exposes to the user and document this properly, that FIFO is not supported from DFS and it's reserved only for dfuse use really for now.
@johannlombardi @bfaccini is this OK to do?
i know lustre has the lmva that doesn't require this special case, but we don't have that in dfs and would add overhead in dir open if we do that.

Reply:
We might eventually have to support VM mounting a DAOS container a root filesystem, so i think that dev/char/fifo files will eventually be required. Another approach would be to have on our own extra flags in DFS to represent links to external container. This could be in addition to IFDIR so maintain compatibility. We could also consider a "lookup + fetch EA" call to do it in a single RPC.
In any case, let's land this patch in the meantime and work on an improved version.

Reply:
I also think that tp deflect inode's mode bits from their original purpose can lead to problems at some point.